### ISSUE TITLE
Native code format checker failing in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#4231]
All PRs are failing with errors like:
```
06:10:33 [native_code_format_check_prtest] $ cmd /c call /tmp/hudson5140982484119790206.bat
06:10:33 FATAL: command execution failed
06:10:33 java.io.IOException: Cannot run program "cmd" (in directory "/mnt/j/workspace/dotnet_corefx/native_code_format_check_prtest"): error=2, No such file or directory
06:10:33       at java.lang.ProcessBuilder.start(ProcessBuilder.java:1047)
```

**Comment by stephentoub** [comment]
@mmitche, @sokket, after #4232, now it appears to be failing for other reasons:
```
08:28:12 Empty changeset, running build.
08:28:12 [native_code_format_check_prtest] $ /bin/sh -xe /tmp/hudson2454304147851405349.sh
08:28:12 + python src/Native/format-code.py checkonly
08:28:12 Clang Format v3.6+ is required
08:28:12 Build step 'Execute shell' marked build as failure
08:28:12 [BFA] Scanning build for known causes...
08:28:12 [BFA] No failure causes found
08:28:12 [BFA] Done. 0s
08:28:12 Setting status of d2e5caa1aa563bb18ccd171f402f11c9ea8f3cf3 to FAILURE with url http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/native_code_format_check_prtest/25/ and message: 'Build finished. No test results found.'
08:28:12 Using context: Code Formatter Check
08:28:13 Finished: FAILURE
```

**Comment by stephentoub** [comment]
We've fixed these issues, but it's now failing because the output when run on Windows (which CI is currently doing) appears to be different from when run on Ubuntu.  After recent fixes, running the script on Ubuntu doesn't yield any differences, whereas running it on Windows yields differences in pal_errno.

**Comment by sokket** [comment]
Investigating...

**Comment by sokket** [comment]
Fixed by #4247
